runtime/src/main/java/org/corfudb/runtime/view/Layout.java

```diff
@@ -319,10 +357,81 @@ public Object clone() throws CloneNotSupportedException {
     }
 
     public enum ReplicationMode {
-        CHAIN_REPLICATION,
-        QUORUM_REPLICATION,
-        REPLEX,
-        NO_REPLICATION
+        CHAIN_REPLICATION {
+
+            @Override
+            public Boolean sealSegment(LayoutSegment layoutSegment, long sealEpoch, CorfuRuntime runtime)
+                    throws QuorumUnreachableException {
+
+                Boolean success;
+
+                for (LayoutStripe layoutStripe : layoutSegment.getStripes()) {
+                    success = false;
+                    CompletableFuture<Boolean>[] logUnitSeaFutures = layoutStripe.getLogServers().stream()
+                            .map(runtime::getRouter)
+                            .map(x -> x.getClient(BaseClient.class))
+                            .map(x -> x.setRemoteEpoch(sealEpoch))
+                            .toArray(CompletableFuture[]::new);
+
+                    QuorumFutureFactory.CompositeFuture<Boolean> quorumFuture =
+                            QuorumFutureFactory.getFirstWinsFuture(Boolean::compareTo, logUnitSeaFutures);
+                    try {
+                        success = quorumFuture.get();
+                    } catch (ExecutionException | InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                    if (!success) throw new QuorumUnreachableException(0, logUnitSeaFutures.length);
+                }
+                return true;
+            }
+        },
+        QUORUM_REPLICATION {
```

Comment:
Why Replex?
In Quorum replication, shouldn't we wait for a quorum of log units to be sealed in every stripe before we can propose a new layout?